### PR TITLE
NUI-937 add semantic release workflow

### PR DESCRIPTION
**Pull request by jdelbick**
## Description
Add semantic release to github action workflow.

Inspired by helix team's [semantic release github action](https://github.com/adobe/htlengine/blob/main/.github/workflows/semantic-release.yaml) and the open source teams example [github action](https://github.com/adobe/starter-repo/blob/master/.github/workflow/npm-publish.yml)

We need the release stage to only run on master, which is less straightfoward than travis and circleci because there is no "branch" environment variable. There are two options described well in this [slack overflow comment](https://stackoverflow.com/a/58142412)
1. Use one workflow and make a condition for the release step to only run on master: Use the [`GITHUB_REF` env variable](https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables) to check if its on master. 
2. Instead of adding a condition to the step, we could make the release be its own github action that only runs on master. However, this gets tricky because we would want the test action to run first but all github actions run in parallel so you would need the run test step in both actions. One action runs on the PR (just the tests) and the other action runs on the merge to master (tests and release). 

Option 2 involves duplicating some code in two places so I thought 1 was a cleaner, more straightforward option.

## Changes
- add step for semantic release
- add "skip" conditions to each step that follow our normal semantic release `[ci skip]` [messages]](https://github.com/adobe/aio-cli-plugin-asset-compute/blob/master/.releaserc.json#L31)
- add semantic release library and config (this repo did not have it before)

Fixes # https://jira.corp.adobe.com/browse/NUI-937

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/node.js-test.yml [hunk 1]
```diff
@@ -30,9 +30,24 @@ jobs:
...
   sizewatcher:
-
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2
     - run: npx @adobe/sizewatcher
+  semantic-release:
+    runs-on: ubuntu-latest
+    needs: [build]
```
Comment (jdelbick) [inline .github/workflows/node.js-test.yml:40]: run only if build phase succeeds (npm test)

File: .github/workflows/node.js-test.yml [hunk 2]
```diff
@@ -30,9 +30,24 @@ jobs:
...
+    if: ${{  !contains(github.event.head_commit.message, '[skip ci]') && github.ref == 'refs/heads/master' }}
```
Comment (jdelbick) [inline .github/workflows/node.js-test.yml:41]: run only if there is no '[skip ci]' in the message & it is the master branch so this only runs on merging to master

File: .github/workflows/node.js-test.yml [hunk 3]
```diff
@@ -30,9 +30,24 @@ jobs:
...
-
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2
     - run: npx @adobe/sizewatcher
+  semantic-release:
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: ${{  !contains(github.event.head_commit.message, '[ci skip]') && github.ref == 'refs/heads/master' }}
```
Comment (jdelbick) [inline .github/workflows/node.js-test.yml:41]: run only if there is no '[skip ci]' in the message & it is the master branch so this only runs on merging to master

see environment variables for explanation of github.ref or GITHUB_REF as an env variable: https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables

File: package.json [hunk 4]
```diff
@@ -35,6 +37,7 @@
   "scripts": {
     "test": "nyc mocha --recursive --exit",
     "posttest": "eslint ./ && license-checker --summary",
-    "beautify": "eslint ./ --fix"
+    "beautify": "eslint ./ --fix",
+    "semantic-release": "semantic-release"
```
Comment (tmathern) [inline package.json:41]: How did you test?

Comment (jdelbick) [reply package.json:41]: yeah that is the issue...I can't test unless I create a dummy repo (hence why I am using this not very important library to test lol)

**Pull request closed without merging**